### DISCUSSION TITLE
Ship canon overheat and jamming

### DISCUSSION THREAD

**Discussion by Zhertva** [FakeFishGames/Barotrauma#5975] [Ideas and suggestions]
Submarine's canons may overheat when fired too quickly or by an unexperienced gunner.
Overheat may lead to damage the guns and jamming the guns, and Mechanical Engineer may have to repair jammed canons from outside the submarine (during combat ! such an epic move !) to make them operational again.
Canons overheat sensibility may be reduced by upgrading canons at the shop, or by using better coils.

**Comment by civilCornball** [comment]
A way to discourage players from spamming the coilgun would, in the long run, aid their skill as a gunner.

I don't think it'd be a good idea to have mechanics head outside the sub during combat, though. Wouldn't the enemies prioritize them, and interrupt the repair by stunning them?

A simple "overheat cooldown" would be best.

**Reply by Nyerske** [reply]
Or engineers not repair it fully, but make it momentarily operational from inside the sub. This way the crew's not fully doomed if it was the only gun on that part of the sub during a big battle. Then when they're out of it the engineer can go out and fully fix it.
